File: Backend-Deploy-Agent/src/models/dto/update_java_project_request_dto.py
from pydantic import BaseModel
from typing import Optional

class UpdateJavaProjectRequestDto(BaseModel):
    id: str
    project_code: Optional[str]
    project_name: Optional[str]
    project_group: Optional[str]
    docker_image_name: Optional[str]
    docker_image_tag: Optional[str]
    external_port: Optional[int]
    internal_port: Optional[int]
    network: Optional[str]
    host_project_path: Optional[str]
    container_project_path: Optional[str]
    git_repository: Optional[str]
    status: Optional[str]

File: Deploy-Center-UI/src/types/dto/UpdateJavaProjectRequestDto.ts
// src/types/UpdateJavaProjectRequestDto.ts
export interface UpdateJavaProjectRequestDto {
  id: string; // 项目的唯一标识符
  project_code?: string;
  project_name?: string;
  project_group?: string;
  docker_image_name?: string;
  docker_image_tag?: string;
  container_name?: string;
  external_port?: number;
  internal_port?: number;
  network?: string;
  jdk_version?: number;
  host_project_path?: string; // 宿主机项目路径
  container_project_path?: string; // 容器项目路径
  git_repository?: string; // Git地址
  // 其他需要更新的字段
}
